src/Extensions/Abstractions/NotNullWhenAttribute.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#if NETFRAMEWORK // This marker attribute only available starting from netstandard 2.1, so we need a replacement to build for full framework
+namespace System.Diagnostics.CodeAnalysis
+{
+	/// <summary>Specifies that when a method returns <see cref="ReturnValue"/>, the parameter will not be null even if the corresponding type allows it.</summary>
+	[AttributeUsage(AttributeTargets.Parameter, Inherited = false)]
+	public sealed class NotNullWhenAttribute : Attribute
+	{
+		/// <summary>
+		/// Initializes the attribute with the specified return value condition
+		/// </summary>
+		/// <param name="returnValue">
```

Comment:
Nitpick: `<param>` tags are typically formatted on their own line.

Reply:
It's a copy of net core class I would keep it as it is